### PR TITLE
cancun-devnet-8 tracker

### PR DESCRIPTION
**Pull request by lightclient**
This PR is tracking the changes for cancun-devnet-8. It is built atop #27511 and has the following changes applied:

* #27849
* changes by me to handle 4788 in the engine api
* a fix from spencer
* #27895
* adding the `shouldOverrideBuilder` tag

### CODE REVIEW

**Comment by holiman** [comment]
> changes by me to handle 4788 in the engine api 

I _think_ those changes are now included in #27872 . PTAL (last 2 commits)

File: core/genesis.go [hunk 1]
```diff
@@ -487,6 +493,17 @@ func (g *Genesis) ToBlock() *types.Block {
...
+				head.BeaconRoot = new(common.Hash)
+			}
+		}
+	}
+	if g.Config != nil && g.Config.IsCancun(big.NewInt(int64(g.Number)), g.Timestamp) {
+		if g.ExcessBlobGas == nil {
+			var excessBlobGas uint64
+			head.ExcessBlobGas = &excessBlobGas
+		} else {
+			head.ExcessBlobGas = g.ExcessBlobGas
```
Comment (fjl) [inline core/genesis.go:506]: This is redundant. `ExcessBlobGas` is already appliedto the header just above.

Comment (fjl) [reply core/genesis.go:506]: Ahh, sorry this branch is outdated. Probably shouldn't review this...

**Comment by lightclient** [comment]
closing this since devnet-8 is finished and most of these changes have made it into master

**Pull request closed without merging**